scrapy/core/downloader/handlers/aiohttp.py

```diff
@@ -0,0 +1,52 @@
+"""Download handlers for http and https schemes"""
+import logging
+import asyncio
+
+import aiohttp
+from twisted.internet import defer
+
+from scrapy.http import Headers
+from scrapy.responsetypes import responsetypes
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPDownloadHandler(object):
+
+    def __init__(self, settings):
+        self.settings = settings
+
+    def download_request(self, request, spider):
+        return _force_deferred(self._download_request(request, spider))
+
+    def _download_request(self, request, spider):
+        """Return a deferred for the HTTP download"""
+        headers=list((k.decode('latin1'), v.decode('latin1'))
+                     for k, vs in request.headers.items()
+                     for v in vs)
+
+        aioresponse = yield from aiohttp.request(
```

Comment:
For overriding/disabling default cookie processing you can override `ClientSession._update_cookie` method -- this is the only place on code which works with session cookies. BTW we don't use cookie jar but `http.cookies.SimpleCookie` cookie dict. `cookiejar` is too cumbersome. Anyway you may use what you want for cookie processing.
`ClientSession` is intended way for aiohttp client, I want to encourage using this one.

Regarding to your feature-list:
> 1- HTTP 1.1 support (http and https)
Done. aiohttp uses 1.1 by default, you may specify 1.0 as `version` parameter. Maybe adding `version` to `ClientSession` ctor makes sense also.
> 2- Connection pooling
> Done via `Connector` objects. You may specify `limit` for maximum concurrency level.
> 3- Keepalive
Done. Response object should be *released* (`yield from resp.release()`, `yield from resp.read()` releases response implicitly).
> 4- Proxy support (socks would be nice too, and sending https urls over plain text too)
http/https proxies are supported, socks can be added by implementing yet another `Connector` subclass. Have no idea what do you mean under *sending https urls over plain text*.
> 5- binding outgoing requests to specific source ips
Please elaborate. Do you mean binding client to specific network interface?
> 6- Stopping downloads by inspecting response headers before downloading whole body (Content-type and content-length checks usually)
Done. Just call `yield from resp.release()` without `yield from resp.read()`.
> 7- Degrade but do not fail to parse invalid responses
It sounds too common. We try to do some tricks for malformed headers but maybe it doesn't satisfy all your needs.
> 8- Whole download timeout, not just connect or read timeouts.
Just use `yield from asyncio.wait_for(session.get(url), timeout=1.5, loop=loop))` :)

Feel free to file missing issues into aiohttp bugtracker. Honestly I appreciate Pull Requests but even just tickets are also very useful for me.
We use aiohttp client for our purposes but massive third party checker as Scrapy is can help with making aiohttp better. So even if you cannot or have no time for making aiohttp PR I'll gladly write it.